std/algorithm/searching.d

```diff
@@ -1204,6 +1210,75 @@ bool endsWith(alias pred, R)(R doesThisEnd)
     }
 }
 
+/**
+Iterates the passed range and selects the extreme element with $(D less).
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+
+Params:
+    map = custom accessor for the comparison key
+    selector = custom mapping for the extrema selection
+    r = Range from which the extreme value will be selected
+
+Returns: The extreme value according to $(D less) of the passed-in values.
+*/
+private auto extremum(alias map = "a", alias selector = "a < b", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias mapFun = unaryFun!map;
+    alias selectorFun = binaryFun!selector;
+
+    // init extremum
+    Unqual!(ElementType!Range) extremeElement = r.front;
+    alias MapType = Unqual!(typeof(mapFun((ElementType!Range).init)));
+    MapType extremeElementMapped = mapFun(r.front);
+
+    static if (isRandomAccessRange!Range && hasLength!Range)
+    {
+        immutable size_t lenArray = r.length;
+        for (size_t i = 1; i < lenArray; i++)
```

Comment:
@ZombineDev I totally agree with you, but I had a look at other functions in this module and a lot of them (e.g. commonPrefix, countUntil, find, findSplit, {min,max}Count, {min,max}Pos, startsWith) had those two cases, so it seems to matter in terms of speed.